runtime/domain.c

```diff
@@ -1724,7 +1741,9 @@ void caml_interrupt_self(void)
 /* async-signal-safe */
 void caml_interrupt_all_signal_safe(void)
 {
-  for (dom_internal *d = all_domains; d < &all_domains[Max_domains]; d++) {
+  for (dom_internal *d = all_domains;
+       d < &all_domains[caml_params->max_domains];
```

Comment:
This is signal-safe as `caml_params->max_domains` is set before signal handlers are installed and it does not change afterwards. It might be good however to comment this fact.

Reply:
Done in https://github.com/ocaml/ocaml/pull/13272/commits/263be361baad53f6ef36f8774ac8b5d9b1572525.